src/main/java/com/liveramp/cascading_ext/yarn/YarnApiHelper.java

```diff
@@ -76,14 +79,62 @@ private static String GETRequest(String urlString) throws IOException {
   }
 
   public static Optional<ApplicationInfo> getYarnAppInfo(Configuration conf, String appId) {
-    String yarnApiAddress = conf.get("yarn.resourcemanager.webapp.address", "");
-    if ( !yarnApiAddress.isEmpty()) {
+    String yarnApiAddress = getYarnApiAddress(conf);
+    if (!yarnApiAddress.isEmpty()) {
       return getYarnAppInfo(yarnApiAddress, appId);
     } else {
       return Optional.empty();
     }
   }
 
+  /***
+   * Gets a yarn api address. This also works if you are using more than one resource manager.
+   * If there's more than one resource manager this method will return the address of the first
+   * resource manager with which it was able to successfully connect to. It will return an empty
+   * string if there were no resource manager it could successfully connect to.
+   */
+  private static String getYarnApiAddress(Configuration conf) {
+    Set<String> yarnApiAddresses = getYarnApiAddresses(conf);
+
+    try {
+      for (String yarnApiAddress : yarnApiAddresses) {
+        if (successfulConnection(yarnApiAddress)) {
+          return yarnApiAddress;
+        }
+      }
+    } catch (IOException e) {
+      LOG.error("Error getting yarn api address:", e);
+    }
+
+    return "";
+  }
+
+  static Set<String> getYarnApiAddresses(Configuration conf) {
+    String emptyString = "";
+    String rmIdsConf = conf.get("yarn.resourcemanager.ha.rm-ids", emptyString);
+
+    if (rmIdsConf.isEmpty()) {
+      String yarnApiAddress = conf.get("yarn.resourcemanager.webapp.address", emptyString);
+      return yarnApiAddress.isEmpty() ? Sets.newHashSet() : Sets.newHashSet(yarnApiAddress);
+    }
+
+    String[] allRmIds = rmIdsConf.split(",");
+    return Arrays.stream(allRmIds)
+        .map(rmId -> conf.get("yarn.resourcemanager.webapp.address." + rmId, emptyString))
+        .filter(address -> !address.isEmpty())
+        .collect(Collectors.toSet());
+  }
+
+  private static boolean successfulConnection(String urlString) throws IOException {
+    URL url = new URL(urlString);
+    HttpURLConnection urlConnection = (HttpURLConnection)url.openConnection();
+    urlConnection.setConnectTimeout(10000);
+    urlConnection.setReadTimeout(10000);
+    int responseCode = urlConnection.getResponseCode();
+
+    return responseCode == 200;
```

Comment:
I actually don't expect that we'd get a 200 here, seems like it's going to redirect you
<img width="828" alt="Screen Shot 2019-10-24 at 5 16 09 PM" src="https://user-images.githubusercontent.com/3924168/67534236-0dd8bc80-f682-11e9-96b1-140557e9f544.png">

we could hit a URL like `http://mstr-4-14-dm-produc-1bf6c8bc-a5cd-4d37-b37c-cb76d4f85d0d.c.liveramp-eng-dm.internal:8088/ws/v1/cluster/info` and see if it's a 200, or check some of the contents to see if it's active/standby or running something

Reply:
good call, I thought I was already doing this.